src/sqlfluff/core/linter/linter.py

```diff
@@ -556,6 +585,21 @@ def lint_parsed(
             # If no parsed tree, set to None
             tree = None
             ignore_buff = []
+            if not parsed.config.get("disable_noqa"):
+                # Templating and/or parsing have failed. Look for "noqa"
+                # comments (the normal path for identifying these comments
+                # requires access to the parse tree, and because of the failure,
+                # we don't have a parse tree).
+                ignore_buff, ignore_violations = cls.extract_ignore_mask_source(
+                    parsed.source_str,
+                    [
+                        lm
+                        for lm in parsed.config.get("dialect_obj").lexer_matchers
+                        if lm.name == "inline_comment"
+                    ][0],
+                    [r.code for r in rule_set],
+                )
+                violations += ignore_violations
```

Comment:
Why are you adding the violations on? Shouldn't you be removing them because they are ignorable violations?

Reply:
This line handles the case where the `noqa` comment is incorrectly formatted. Basically, if the "noqa" comments don't "parse" correctly.

The important thing here is that we're assigning to `ignore_buffer`. That is the list of parsed "noqa" comments. It'll be applied later. See line 569 in this same file -- that's the normal workflow for assigning to `ignore_buffer`. There are two cases:
* Normal case: Successful template and parse (line 569)
* Error case: Templating and/or parsing fails (here)

In each case, we assign to `ignore_buff`, which is the list of "noqa`" directives. They are actually applied later. Here, all we need to do is build that list and pass it on.